com.unity.render-pipelines.core/Runtime/Textures/Texture2DAtlas.cs

```diff
@@ -290,35 +295,49 @@ protected bool Is2D(Texture texture)
+        /// <returns>true on single channel conversion false otherwise</returns>
+        protected bool IsSingleChannelBlit(Texture source, Texture destination)
         {
-            int mipCount = GetTextureMipmapCount(texture.width, texture.height);
-
-            if (!blitMips)
-                mipCount = 1;
-
-            for (int mipLevel = 0; mipLevel < mipCount; mipLevel++)
+            var srcCount = GraphicsFormatUtility.GetComponentCount(source.graphicsFormat);
+            var dstCount = GraphicsFormatUtility.GetComponentCount(destination.graphicsFormat);
+            if (srcCount == 1 || dstCount == 1)
             {
-                cmd.SetRenderTarget(m_AtlasTexture, mipLevel);
-                Blitter.BlitQuad(cmd, texture, sourceScaleOffset, scaleOffset, mipLevel, true);
+                // One to many, many to one
+                if (srcCount != dstCount)
+                    return true;
+
+                // Single channel swizzle
+                var srcSwizzle =
+                    ((1 << ((int)GraphicsFormatUtility.GetSwizzleA(source.graphicsFormat) & 0x7)) << 24) |
+                    ((1 << ((int)GraphicsFormatUtility.GetSwizzleB(source.graphicsFormat) & 0x7)) << 16) |
+                    ((1 << ((int)GraphicsFormatUtility.GetSwizzleG(source.graphicsFormat) & 0x7)) << 8) |
+                    ((1 << ((int)GraphicsFormatUtility.GetSwizzleR(source.graphicsFormat) & 0x7)));
+                var dstSwizzle =
+                    ((1 << ((int)GraphicsFormatUtility.GetSwizzleA(destination.graphicsFormat) & 0x7)) << 24) |
+                    ((1 << ((int)GraphicsFormatUtility.GetSwizzleB(destination.graphicsFormat) & 0x7)) << 16) |
+                    ((1 << ((int)GraphicsFormatUtility.GetSwizzleG(destination.graphicsFormat) & 0x7)) << 8) |
+                    ((1 << ((int)GraphicsFormatUtility.GetSwizzleR(destination.graphicsFormat) & 0x7)));
+                if (srcCount == dstCount &&
+                    srcSwizzle != dstSwizzle)
+                    return true;
             }
+
+            return false;
         }
 
-        /// <summary>
-        /// Blit and project Cube texture into an octahedral 2D texture in the atlas.
-        /// </summary>
-        /// <param name="cmd">Target command buffer for graphics commands.</param>
-        /// <param name="scaleOffset">Destination scale (.xy) and offset (.zw)</param>
-        /// <param name="texture">Source Texture</param>
-        /// <param name="blitMips">Blit mip maps.</param>
-        protected void BlitCubeTexture2DOctahedral(CommandBuffer cmd, Vector4 scaleOffset, Texture texture, bool blitMips = true)
+        private enum BlitType
+        {
+            Default,
+            CubeTo2DOctahedral,
+            SingleChannel,
+            CubeTo2DOctahedralSingleChannel,
+        }
+
+        private void Blit2DTexture(CommandBuffer cmd, Vector4 scaleOffset, Texture texture, Vector4 sourceScaleOffset, bool blitMips, BlitType blitType)
```

Comment:
There has been only potential internal usage before. The atlases have been moved to here and have not yet been released. 
The behavior of the function has also changed. I just reused the name. Now it's collects all duplicate loops into this one.

Reply:
I don't think there's any non-beta code shipped. Yeah, so these come from HDRP. I had to make more changes that I expected. I think the API is a bit more consistent now. I've tried to keep as much as I can as they were (internal), but most usecases are outside of .core package/assembly, so that forces the API public.
I suppose some things could be internal protected.
I can do one more pass over the methods.

Reply:
I reduced the visibility of some of the fields and methods. All public methods (or their overload) has a use.